src/persist/src/operators/stream.rs

```diff
@@ -0,0 +1,174 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! A Timely Dataflow operator that passes through its input after persisting
+//! it.
+
+use std::convert::identity;
+use std::sync::mpsc;
+
+use timely::dataflow::operators::{Concat, OkErr};
+use timely::dataflow::{Scope, Stream};
+use timely::Data;
+
+use crate::error::Error;
+use crate::indexed::runtime::{StreamReadHandle, StreamWriteHandle};
+use crate::operators;
+use crate::Token;
+
+/// A Timely Dataflow operator that passes through its input after persisting
+/// it.
+pub trait PersistentStream<G: Scope<Timestamp = u64>, K: Data, V: Data> {
```

Comment:
so looking at other timely operators like `filter` ( https://docs.rs/timely/0.12.0/timely/dataflow/operators/filter/trait.Filter.html ) I think maybe this trait should be named `Persist` and the function should be named `persist`?

otherwise I think this risks confusing people because a natural interpretation here is "is this like the `Stream` struct in timely but persisted?" (that was my first gut idea when I started reading)

Reply:
Love it! Done